data/regions/US.yml

```diff
@@ -10,7 +10,7 @@ group_name: North America
 name_alternates:
 - United States of America
 zip_label: Zip code
-zip_regex: "(?-mix:\\A\\d{5}(-\\d{4})?\\z)|(?i-mx:\\AFPO\\s*A[A-Z]\\s*\\d{5}-\\d{4}\\z)"
+zip_regex: "(?-mix:\\A\\d{5}(-\\d{4})?\\z)"
```

Comment:
Quoting prior work from @cejaekl:

We noticed that our US regex was defective, and have corrected it. The old regex seems to have been based on a misunderstanding about how the USA Fleet Post Office (FPO) addressing works.

Here's an example FPO address (from https://pe.usps.com/text/pub28/28c2_019.htm):
```
PVT MIKE JONES
PSC 3022 BOX 684
FPO AE 09397-0007
UNITED STATES
```
In this example, the values should be entered into address fields as follows:
 - `name`: `PVT MIKE JONES`
 - `address1`: `PSC 3022 BOX 684`
 - `city`: `FPO` (Fleet Post Office)
 - `province`: `AE` (pseudo-state code for Armed forces, Europe)
 - `zip`: `09397-0007`
 - `country`: `UNITED STATES`

The old regex was permitting values such as
`FPO AE 09397-0007` in the `zip` field.